chips/lowrisc/src/i2c.rs

```diff
@@ -0,0 +1,494 @@
+        (0x3C => timing4: ReadWrite<u32, TIMING4::Register>),
+        (0x40 => timeout_ctrl: ReadWrite<u32, TIMEOUT_CTRL::Register>),
+        (0x44 => @END),
+    }
+}
+
+register_bitfields![u32,
+    INTR [
+        FMT_WATERMARK OFFSET(0) NUMBITS(1) [],
+        RX_WATERMARK OFFSET(1) NUMBITS(1) [],
+        FMT_OVERFLOW OFFSET(2) NUMBITS(1) [],
+        RX_OVERFLOW OFFSET(3) NUMBITS(1) [],
+        NAK OFFSET(4) NUMBITS(1) [],
+        SCL_INTERFERENCE OFFSET(5) NUMBITS(1) [],
+        SDA_INTERFERENCE OFFSET(6) NUMBITS(1) [],
+        STRETCH_TIMEOUT OFFSET(7) NUMBITS(1) [],
+        SDA_UNSTABLE OFFSET(8) NUMBITS(1) []
+    ],
+    CTRL [
+        ENABLEHOST OFFSET(0) NUMBITS(1) []
+    ],
+    STATUS [
+        FMTFULL OFFSET(0) NUMBITS(1) [],
+        RXFULL OFFSET(1) NUMBITS(1) [],
+        FMTEMPTY OFFSET(2) NUMBITS(1) [],
+        HOSTIDLE OFFSET(3) NUMBITS(1) [],
+        TARGETIDLE OFFSET(4) NUMBITS(1) [],
+        RXEMPTY OFFSET(5) NUMBITS(1) []
+    ],
+    RDATA [
+        RDATA OFFSET(0) NUMBITS(8) []
+    ],
+    FDATA [
+        FBYTE OFFSET(0) NUMBITS(8) [],
+        START OFFSET(8) NUMBITS(1) [],
+        STOP OFFSET(9) NUMBITS(1) [],
+        READ OFFSET(10) NUMBITS(1) [],
+        RCONT OFFSET(11) NUMBITS(1) [],
+        NAKOK OFFSET(12) NUMBITS(1) []
+    ],
+    FIFO_CTRL [
+        RXRST OFFSET(0) NUMBITS(1) [],
+        FMTRST OFFSET(1) NUMBITS(1) [],
+        RXILVL OFFSET(2) NUMBITS(3) [
+            RXLVL1 = 0,
+            RXLVL4 = 1,
+            RXLVL8 = 2,
+            RXLVL16 = 3,
+            RXLVL30 = 4
+        ],
+        FMTILVL OFFSET(5) NUMBITS(3) [
+            FMTLVL1 = 0,
+            FMTLVL4 = 1,
+            FMTLVL8 = 2,
+            FMTLVL16 = 3,
+            FMTLVL30 = 4
+        ]
+    ],
+    FIFO_STATUS [
+        FMTLVL OFFSET(0) NUMBITS(6) [],
+        RXLVL OFFSET(16) NUMBITS(6) []
+    ],
+    OVRD [
+        TXOVRDEN OFFSET(0) NUMBITS(1) [],
+        SCLVAL OFFSET(1) NUMBITS(1) [],
+        SDAVAL OFFSET(2) NUMBITS(1) []
+    ],
+    VAL [
+        SCL_RX OFFSET(0) NUMBITS(1) [],
+        SDA_RX OFFSET(1) NUMBITS(1) []
+    ],
+    TIMING0 [
+        THIGH OFFSET(0) NUMBITS(16) [],
+        TLOW OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING1 [
+        T_R OFFSET(0) NUMBITS(16) [],
+        T_F OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING2 [
+        TSU_STA OFFSET(0) NUMBITS(16) [],
+        THD_STA OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING3 [
+        TSU_DAT OFFSET(0) NUMBITS(16) [],
+        THD_DAT OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING4 [
+        TSU_STO OFFSET(0) NUMBITS(16) [],
+        T_BUF OFFSET(16) NUMBITS(16) []
+    ],
+    TIMEOUT_CTRL [
+        VAL OFFSET(0) NUMBITS(31) [],
+        EN OFFSET(31) NUMBITS(1) []
+    ]
+];
+
+pub struct I2c<'a> {
+    registers: StaticRef<I2cRegisters>,
+    clock_period_nanos: u32,
+
+    master_client: OptionalCell<&'a dyn hil::i2c::I2CHwMasterClient>,
+
+    // Set whe calling the write_read operation
```

Comment:
```suggestion
    // Set when the write_read operation is called
```

Reply:
Actually, I don't think I quite understand what `address` this is referring to. Is this the address of this device when it's acting as a slave? Or is this the address on the slave device that's being read from? (I think the latter?) A clearer name might help here.

Reply:
I have changed it to `slave_read_address` and added a better comment.